operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -388,50 +390,55 @@ func (s *RedpandaControllerSuite) TestClusterSettings() {
 			},
 		},
 		{
+			// Adding a test case with data_transforms_enabled which requires a
+			// cluster restart, to validate that the operator ensures the cluster
+			// is restarted.
+			Name: "should_enable_transforms",
 			In: map[string]any{
 				"data_transforms_enabled": true, // needs restart
 			},
 			Expected: map[string]any{
 				"data_transforms_enabled": true,
 			},
-			ExpectedRestart: true,
 		},
 	}
 
 	for _, c := range cases {
-		adminClient, err := s.clientFactory.RedpandaAdminClient(s.ctx, rp)
-		s.Require().NoError(err)
-		defer adminClient.Close()
-		st, err := adminClient.ClusterConfigStatus(s.ctx, false)
-		assert.NoError(s.T(), err)
-		initialVersion := slices.MaxFunc(st, func(a, b rpadmin.ConfigStatus) int {
-			return int(a.ConfigVersion - b.ConfigVersion)
-		}).ConfigVersion
-
-		waitFn := setConfig(c.In)
-		s.EventuallyWithT(func(t *assert.CollectT) {
+		s.Run(fmt.Sprintf(c.Name, c.In), func() {
```

Comment:
I've added a name to the test cases to make it easier to identify what case is failing